<?xml version="1.0"?>

<launch>


    <arg name="num_of_bots" default="12"/>

    <!-- Map server -->
    <arg name="map_file" default="$(find local_sensing_intest)/map/map.yaml"/>
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)"/>
<!--    <node name="map_point_cloud" pkg="local_sensing_intest" type="grid_to_pointcloud2.py"/>-->

    <param name="initial_pose/x" type="double" value="0"/>
    <param name="initial_pose/y" type="double" value="0"/>
    <param name="initial_pose/a" type="double" value="0"/>

    <!--     Transorms -->
    <include file="$(find local_sensing_intest)/launch/transforms_replay.launch">
        <arg name="num_of_bots" value="$(arg num_of_bots)" />
        <arg name="fixed_frame_origin_tag" value="$(arg num_of_bots)" />
        <arg name="odom_est_type" value="rigid_joint" />
    </include>

    <!--     Set initial pose using AprilTags -->
    <!--    <node name="set_initial_pose" pkg="local_sensing_intest" type="initial_pose_from_apriltags.py" respawn="false"/>-->

    <!-- Rosbag play -->
    <node pkg="rosbag" type="play" name="player" output="screen" args="--clock $(env bag_file) -r 1"/>

    <!-- PointCloud -->
    <node name="point_cloud2_broadcaster" pkg="local_sensing_intest" type="point_cloud2_broadcaster_replay.py"
          respawn="true" output="screen" args="$(arg num_of_bots)"/>
    <node name="pc_frame_transform" pkg="local_sensing_intest" type="pc_frame_transform.py"
          respawn="true" output="screen"/>
    <node name="pc_feature_extractor" pkg="local_sensing_intest" type="pcl_feature_ext.py" respawn="true"
          output="screen"/>

    <!--    PointCloud localization-->
<!--    <node pkg="local_sensing_intest" type="pcl_registeration.py" name="pcl_registration" output="screen"-->
<!--          respawn="true"/>-->

    <!-- PointCloud to LaserScan -->
    <include file="$(find local_sensing_intest)/launch/point_cloud_to_laserscan.launch"/>

    <!-- AMCL -->
    <include file="$(find local_sensing_intest)/launch/amcl.launch"/>

    <!-- Odometry -->
    <node name="odometry" pkg="local_sensing_intest" type="odometry.py" respawn="true" output="screen"/>

    <!--    IMU_frame_converter-->
    <node name="imu_frame_conversion" pkg="local_sensing_intest" type="imu_frame_conversion.py" respawn="true" args="12"
          output="screen"/>

    <!--     Robot localization package -->
<!--    <include file="$(find local_sensing_intest)/launch/robot_loc_pkg.launch"/>-->

    <!--    Rviz-->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find local_sensing_intest)/rviz/localization.config.rviz"
          respawn="true"/>

    <!--    Save csv data-->
    <node type="csv_saver.py" name="csv_saver" pkg="local_sensing_intest" respawn="true" output="screen"/>


    <!--    Record data-->
    <!--    <node pkg="rosbag" type="record"-->
    <!--          args="-o /home/amin/catkin_ws/src/local_sensing_intest/bags/ bot00/imu/data bot00/dist/data bot01/imu/data-->
    <!--                  bot01/dist/data bot02/imu/data bot02/dist/data bot03/imu/data bot03/dist/data bot04/imu/data bot04/dist/data-->
    <!--                  bot05/imu/data bot05/dist/data bot06/imu/data bot06/dist/data bot07/imu/data bot07/dist/data bot08/imu/data-->
    <!--                  bot08/dist/data bot09/imu/data bot09/dist/data bot10/imu/data bot10/dist/data bot11/imu/data bot11/dist/data-->
    <!--                  state map tf cloud2 control_cmd locomotion_stats"-->
    <!--          name="record"/>-->

</launch>
